File: simple_problems/fizzbuzz.py
"""
Wording:
-------
Write a program that outputs the string representation of numbers from 1 to n.

But for multiples of 3 it should output “Fizz” instead of the number and for the multiples of 5 output “Buzz”. For numbers which are multiples of both three and five output “FizzBuzz”.
"""

from typing import List


def fizzbuzz(n: int) -> List[str]:
    output = []

    if n < 1:
        return output

    else:
        for i in range(1, n + 1):
            if i % 3 == 0 or i % 5 == 0:
                output.append(
                    (i % 3 == 0) * "Fizz" + (i % 5 == 0) * "Buzz"
                )
            else:
                output.append(str(i))
        
        return output


if __name__ == '__main__':
    pass

File: classic-puzzles/puzzles-with-modulo/fizzbuzz.js
"use strict";

/*
Wording:
-------
Write a program that outputs the string representation of numbers from 1 to n.

But for multiples of 3 it should output “Fizz” instead of the number and for the multiples of 5 output “Buzz”. For numbers which are multiples of both three and five output “FizzBuzz”.
*/

/**
 * @param {number} n 
 * @returns {Array.string}
 */
function fizzbuzz(n) {
  let output = [];

  if (n < 1) {
    return output;
  } else {
    for (let i = 1; i <= n; i++) {
      if (i % 15 === 0) {
        output.push("FizzBuzz");
      } else if (i % 3 === 0) {
        output.push("Fizz");
      } else if (i % 5 === 0) {
        output.push("Buzz");
      } else {
        output.push(String(i));
      }
    }
    return output;
  }
}

module.exports = fizzbuzz;
